src/lib.rs

```diff
@@ -224,30 +220,39 @@ pub fn server_login_start(
     )
     .map_err(from_protocol_error("start serverLogin"))?;
 
-    let credential_response = BASE64.encode(server_login_start_result.message.serialize());
-    let server_login = BASE64.encode(server_login_start_result.state.serialize());
+    let login_response = BASE64.encode(server_login_start_result.message.serialize());
+    let server_login_state = BASE64.encode(server_login_start_result.state.serialize());
 
-    let result = ServerLoginStartResult {
-        server_login,
-        credential_response,
+    let result = StartServerLoginResult {
+        server_login_state,
+        login_response,
     };
     Ok(result)
 }
 
 #[derive(Debug, Serialize, Deserialize, Tsify)]
 #[tsify(into_wasm_abi, from_wasm_abi)]
-pub struct ServerLoginFinishParams {
-    #[serde(rename = "serverLogin")]
-    server_login: String,
-    #[serde(rename = "credentialFinalization")]
-    credential_finalization: String,
+pub struct FinishServerLoginParams {
+    #[serde(rename = "serverLoginState")]
+    server_login_state: String,
+    #[serde(rename = "finishLoginRequest")]
+    finish_login_request: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, Tsify)]
+#[tsify(into_wasm_abi, from_wasm_abi)]
+pub struct FinishServerLoginResult {
+    #[serde(rename = "sessionKey")]
+    session_key: String,
 }
 
-#[wasm_bindgen(js_name = serverLoginFinish)]
-pub fn server_login_finish(params: ServerLoginFinishParams) -> Result<String, JsError> {
+#[wasm_bindgen(js_name = finishServerLogin)]
+pub fn finish_server_login(
+    params: FinishServerLoginParams,
+) -> Result<FinishServerLoginResult, JsError> {
     let credential_finalization_bytes =
-        base64_decode("credentialFinalization", params.credential_finalization)?;
-    let state_bytes = base64_decode("serverLogin", params.server_login)?;
+        base64_decode("credentialFinalization", params.finish_login_request)?;
```

Comment:
also here, shouldn't this be `finishLoginRequest` instead of `credentialFinalization`?

Reply:
yeah there will be a couple more probably. I'll have to do another pass 😅